modin/core/dataframe/pandas/dataframe/dataframe.py

```diff
@@ -2675,15 +2675,17 @@ def broadcast_apply_full_axis(
             enumerate_partitions=enumerate_partitions,
             keep_partitioning=keep_partitioning,
         )
-        # Index objects for new object creation. This is shorter than if..else
-        kw = self.__make_init_labels_args(new_partitions, new_index, new_columns)
+        kw = {}
         if dtypes == "copy":
             kw["dtypes"] = self._dtypes
-        elif dtypes is not None:
+        elif dtypes is not None and new_columns is not None:
+            columns = new_columns if new_columns is not None else self.columns
```

Comment:
This branch will continue to trigger the calculation of the columns, but this is not important for `pivot_table`, since we will not get into this branch.

Reply:
I would prefer not to discard the `dtypes` hint in this case, we would better trigger columns computation in this case. As `pivot_table` doesn't provide `dtypes` anyway then discarding these changes won't affect the case you're trying to optimize :)

As a proper laziness, we may want to store the scalar `dtypes` hint in `self._dtype` and materialize the properly-shaped series on demand, but let's leave it for the future.

Reply:
you probably saw the previous version of the code :)

Reply:
I don't get it. What's the previous version?

Reply:
>I would prefer not to discard the dtypes hint in this case, we would better trigger columns computation in this case. As pivot_table doesn't provide dtypes anyway then discarding these changes won't affect the case you're trying to optimize :)

The code already works as you wrote, I assumed you didn't update the pull request before the comment.
```python
columns = new_columns if new_columns is not None else self.columns
...
[np.dtype(dtypes)] * len(columns), index=columns
```